src/Components/Registration/registrationForm.js

```diff
@@ -0,0 +1,107 @@
+import React, { useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { Button, Form, Spinner } from "react-bootstrap";
+import "./registrationForm.css";
+
+import { RegistationRequest } from "../../Redux/createAction/createAction";
+import { useHistory } from "react-router-dom";
+
+const Registration = () => {
+  const [username, setusername] = useState("");
+  const [password, setpassword] = useState("");
+  const [option, setoptions] = useState("Admin");
+  const history = useHistory();
+
+  const dispatch = useDispatch();
+  const state = useSelector((state) => {
+    return state;
+  });
+
+  const regisstatus = state.Regis_status;
+  const handlesubmit = () => {
+    let formData = {
+      username: username,
+      password: password,
+      option: option,
+    };
+    dispatch(RegistationRequest(formData));
+    setusername("");
+    setpassword("");
+    if (regisstatus.isLoading === false && regisstatus.isRegistered === false) {
+    }
+  };
+  if (regisstatus.isRegistered) {
+    history.push("./login");
+  }
+  return (
+    <React.Fragment>
+      <div className="reg">
+        <h2>Signup</h2>
+      </div>
+
+      <div className="regis">
+        <Form>
+          <Form.Group>
+            <Form.Label>Username</Form.Label>
+            <Form.Control
+              type="text"
+              placeholder="Enter your username"
+              value={username}
+              onChange={(e) => setusername(e.target.value)}
+            />
+          </Form.Group>
+          <Form.Group controlId="formGroupPassword">
+            <Form.Label>Password</Form.Label>
+            <Form.Control
+              type="password"
+              placeholder="Password"
+              value={password}
+              onChange={(e) => setpassword(e.target.value)}
+            />
+          </Form.Group>
+
+          <Form.Group controlId="formGridState">
+            <Form.Label>Select user type</Form.Label>
+            <Form.Control
+              name={option}
+              as="select"
+              onChange={(e) => setoptions(e.target.value)}
+            >
+              <option value="Admin">Admin</option>
+              <option value="Guest">Guest User</option>
+            </Form.Control>
+          </Form.Group>
+          <Button
+            variant="primary"
+            disabled={(username.length && password.length) === 0}
```

Comment:
we can handle this with some other logic, === 0 is not good

Reply:
sir can you suggest?should i use global state  or local state for disabling the button
i am currently using local state

Reply:
inital state will null on type it will change then you 
!(username && password) this i s enough

Reply:
done